src/telemetry.c

```diff
@@ -401,6 +401,64 @@ static int set_timestamp_header(struct telem_ref *t_ref)
 }
 
 /**
+ * Sets cpu model for telemetry record. The information from cpu is extracted
+ * from /proc/cpuinfo, specifically "model name" attribute.
+ *
+ * @param t_ref Telemetry Record reference obtained from tm_create_record.
+ *
+ * @return 0 if successful, or a negative errno-style value if not.
+ *
+ */
+static int set_cpu_model_header(struct telem_ref *t_ref)
+{
+        FILE *fs = NULL;
+        char buf[SMALL_LINE_BUF] = { 0 };
+        char *model_name = NULL;
+        const char *attr_name = "model name";
+        int status = 0;
+        size_t attr_len = sizeof(attr_name);
```

Comment:
should be strlen(attr_name)? sizeof() will return the pointer size in this case...